### ISSUE TITLE
Long idle time during production build process.

### ISSUE THREAD

**Issue by wasong** [webpack/webpack#15363]
# Bug report

**What is the current behavior?**
Upgrading to latest Webpack v5 from v4 has increased production build times and culprit seems to be an `idle` process.
Cold build time with v5 is around 70 - 80s and rebuilds are also 70 - 80s.
Previously in v4, cold builds were around 45s and rebuilds were between 10-20s.

**If the current behavior is a bug, please provide the steps to reproduce.**
Repo is available here: https://github.com/wasong/effective-guacamole
I've stripped it to the bare minimum but left in the packages because I'm not sure what's causing it.
I've also included the cpuprofile data in the public folder for visibility. The 2 `profile-real-*` is from my app with all the files. The other 2 are generated from the issue repo.

You can see in the profiles that there is an `idle` process that doesn't close for an extended period of time.

**What is the expected behavior?**
Expected to build in similar time as v4.

**Other relevant information:**
webpack version:  5.68.0
Node.js version: 16.14.0
Operating System: macos monterey 12.3 beta
Additional tools:

**Comment by markjm** [comment]
So the profiles here can be a little confusing because (as you pointed out), there is just a large gap where the profile is just idle. The reason for this gap is because terser-webpack-plugin runs in worker threads, so you will just see that gap in the profile. The long idle does look to be due to terser actively minifying your code (and not some errant idle/sleep). You can get a better looking trace by turning off workers for terser (not recommended for your actual app, but maybe it will help you with profiling)

```
    optimization: {
        minimize: true,
        minimizer: [
            new TerserPlugin({
                parallel: false,
            }),
        ],
```

Seems like all the time is taken up minimizing `@babel/polyfill` - maybe there are some golden nuggets in the profiles to find optimizations there, but it is a fair bit of code in there for terser to chug through

**Comment by vankop** [comment]
regarding terser webpack@5 by default has 2 passes, so technically minimizing in webpack@5 should take more time..

**Comment by wasong** [comment]
Back with an update!

So I found out I can remove babel-polyfill and use alternative solutions so I was able to clear that from my dependencies.
I didn't see a change in build time until I looked into what @vankop had said.

I found through [here](https://webpack.js.org/blog/2020-10-10-webpack-5-release/) that terser does 2 passes. I set it to minimize with one pass and I was able to achieve a similar performance as v4. Zero change rebuilds we're also near-instant whereas with 2 passes, it seemed like it would rebuild from scratch.

Is it necessary to do two passes with terser?
